hubverse_annotator/app.py

```diff
@@ -415,94 +522,75 @@ def load_data_ui() -> tuple[pl.DataFrame, pl.DataFrame]:
         forecast table or an empty DataFrame.
     """
     observed_data_file = st.file_uploader(
-        "(Optional) Upload Hubverse Target Data", type=["parquet"]
+        "Upload Hubverse Target Data", type=["parquet"]
     )
-    observed_data_table = load_hubverse_table(observed_data_file)
-    if (
-        not observed_data_table.is_empty()
-        and "as_of" in observed_data_table.columns
-    ):
-        latest = observed_data_table.select(pl.col("as_of").max()).item()
-        observed_data_table = observed_data_table.filter(
-            pl.col("as_of") == latest
+    if observed_data_file:
+        observed_data_table = load_hubverse_table(observed_data_file)
+        if "as_of" in observed_data_table.columns:
+            observed_data_table = observed_data_table.filter(
+                pl.col("as_of") == pl.col("as_of").max()
+            )
+    else:
+        observed_data_table = pl.DataFrame(
+            schema={
+                "date": pl.Date,
+                "state": pl.Utf8,
+                "observation": pl.Float64,
+                "location": pl.Utf8,
+                "as_of": pl.Date,
+                "target": pl.Utf8,
+            }
```

Comment:
You have already define `load_hubverse_table(None)`, so this could be made redundant. I think it is a bit more elegant to move this empty dataframe piece inside `load_hubverse_table`.